### ISSUE TITLE
False positive for one_member_abstracts

### ISSUE THREAD

**Issue by lexaknyazev** [dart-lang/linter#990]
Docs say:
> Avoid defining a one-member abstract class when a simple function will do.

I don't think there's a way of avoiding one-member abstracts when they are used as "interfaces" like this:
```dart
abstract class Fooable { // one_member_abstracts 
  void foo();
}

void main() {
  Object o;
  if (o is Fooable) {
    o.foo();
  }
}
```

**Comment by munificent** [comment]
Yeah, I'm not surprised there are false positives. It is an "AVOID" and not a "DON'T" rule, after all.

**Comment by pq** [comment]
I think this is working as intended and a case where a judicious `//ignore` is the right solution.  (Ideally with a doc comment.)
